src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,66 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeptions.InvalidUserAgeException;
+import core.basesyntax.exeptions.InvalidUserDataException;
+import core.basesyntax.exeptions.InvalidUserLoginException;
+import core.basesyntax.exeptions.InvalidUserPasswordException;
+import core.basesyntax.exeptions.NullUserException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new NullUserException();
+        } else if (isUserValid(user)) {
+            storageDao.add(user);
+            return user;
+        } else {
+            return null;
```

Comment:
Returning `null` when user validation fails is not ideal. Consider throwing an `InvalidUserDataException` or another specific exception to indicate why registration failed, which would make error handling more explicit.